Make playbin buffer-size and buffer-duration configurable in mopidy/audio section (#1950)
mopidy/mopidy: docs/config.rst

    Expects an integer above 0.

    Sets the ``max-size-time`` property of the GStreamer queue. If you experience buffering
    before track changes, it may help to increase this, possibly by at least a
    few seconds. The default is letting GStreamer decide the size, which at the
    time of this writing is 1000.

.. confval:: audio/min_buffer_size

    Sets the ``buffer-size`` property of the GStreamer playbin, in bytes.

    Expects an integer 0 or higher. Defaults to 5242880.

    Sets the minimum buffer size of the GStreamer playbin. If you experience
    buffering/stuttering, it may help to increase this. If you experience
    latency when changing tracks, lower this value.
    If set to zero, the minimum buffer size will be governed by :confval:`audio/min_buffer_duration` instead.

.. confval:: audio/min_buffer_duration

>     The minimum buffer duration that is given to GStreamer playbin, in milliseconds.

Reviewer: This one needed changing in the exact same way as the ones above
```suggestion
    Sets the ``buffer-duration`` property of the GStreamer playbin, in milliseconds.
```

    Expects an integer 0 or higher. Defaults to 5000.

    Sets the minimum buffer duration of the GStreamer playbin. If you experience
    buffering/stuttering, it may help to increase this. If you experience latency
    when changing tracks, lower this value.

>     A value of 0 seems can work fine, although we recommend a minimum value of
>     250ms so small internet disturbances will not disturb your audio stream.

Reviewer: ```suggestion
```

    Expects an integer above 0.

    Sets the maximum buffer size of the GStreamer queue. If you experience buffering
    before track changes, it may help to increase this, possibly by at least a
    few seconds. The default is letting GStreamer decide the size, which at the
    time of this writing is 1000.

.. confval:: audio/min_buffer_size

    The minimum buffer size that is given to GStreamer playbin, in bytes.

    Expects an integer 0 or higher.

    Sets the minimum buffer size of the GStreamer playbin. If you experience
    buffering/stuttering, it may help to increase this. If you experience
    latency when changing tracks, lower this value.
    A value of 0 seems to work fine, then the minimum buffer size seems to be

>     determined by min_buffer_duration.

Reviewer 2: I think this sentence is a bit confusing, especially since the default value isn't zero.

    The default is 5242880 bytes, which was the old hardcoded value

>     (mopidy versions 3.1.0 or lower).

Reviewer 2: Isn't that still the default value?

.. confval:: audio/min_buffer_duration

    The minimum buffer duration that is given to GStreamer playbin, in milliseconds.

    Expects an integer 0 or higher.

    Sets the minimum buffer duration of the GStreamer playbin. If you experience
    buffering/stuttering, it may help to increase this. If you experience latency
    when changing tracks, lower this value.

>     A value of 0 seems can work fine, although we recommend a minimum value of

Reviewer 2: You should change this:

> A value of 0 seems can work fine

To something like:

> A value of 0 can work fine

    250ms so small internet disturbances will not disturb your audio stream.
    The default is 5000 milliseconds, which was the old hardcoded value

>     (mopidy versions 3.1.0 or lower).

Reviewer 2: As far as I can tell this is still the default value.

Author: I would expect a few developers to test out the values I am suggesting, and if the outcome is consistently better (which I expect) to take these as the new default values....
